eth/client.go

```diff
@@ -876,7 +878,13 @@ func (c *client) LatestBlockNum() (*big.Int, error) {
 func (c *client) ProcessHistoricalUnbond(startBlock *big.Int, cb func(*contracts.BondingManagerUnbond) error) error {
 	// Retrieve historical logs starting from startBlock
 	// WatchForUnbond() will not emit past logs
-	filterOpts := &bind.FilterOpts{Start: startBlock.Uint64()}
+	ctx, cancel := context.WithTimeout(context.Background(), RPCTimeout)
+	defer cancel()
+
+	filterOpts := &bind.FilterOpts{
+		Start:   startBlock.Uint64(),
+		Context: ctx,
+	}
```

Comment:
Could these be factored out into a helper with the sig `func (startBlock *big.Int) *bind.FilterOpts` ?

Reply:
Updated to use a `filterOptsWithTimeout()` helper